pythonPackages.{pyshark,pylxd,nsenter}: init

petabyteboy opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change

This adds some general purpose python modules that I need to nixpkgs.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

jonringer commented on pkgs/development/python-modules/argparse/default.nix:
```diff
@@ -0,0 +1,11 @@
+{ buildPythonPackage, fetchPypi }:
+
+buildPythonPackage rec {
+  pname = "argparse";
+  version = "1.4.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1r6nznp64j68ih1k537wms7h57nvppq0szmwsaf99n71bfjqkc32";
+  };
+}
```
this is in the stdlib, it should not be added

petabyteboy replied:
Has been removed from the PR

jonringer commented on pkgs/development/python-modules/cherrypy/default.nix:
```diff
@@ -41,6 +41,8 @@ buildPythonPackage rec {
     objgraph pytest pytestcov pathpy requests_toolbelt pytest-services
   ];
 
+  doCheck = false;
```
why?

petabyteboy replied:
Because the failing tests were preventing me from building pylxd. I took a closer look now and found that it was only one test, so I deselected that. The breakage started with the python 3.7.6 -> 3.7.7 update.

jonringer commented on pkgs/development/python-modules/nsenter/default.nix:
```diff
@@ -0,0 +1,25 @@
+
+  propagatedBuildInputs = [
+    contextlib2
+    argparse
```
```suggestion
```

petabyteboy replied:
done

jonringer commented on pkgs/development/python-modules/nsenter/default.nix:
```diff
@@ -0,0 +1,25 @@
+    sed '/^pathlib$/d' -i requirements.txt # included in std since python 3.4
+  '';
+
+  doCheck = false;
+}
```
add meta section

petabyteboy replied:
done

jonringer commented on pkgs/development/python-modules/nsenter/default.nix:
```diff
@@ -0,0 +1,25 @@
+    sed '/^pathlib$/d' -i requirements.txt # included in std since python 3.4
+  '';
+
+  doCheck = false;
```
why?

petabyteboy replied:
Added a comment on why. The tarball from PyPi does not contain any tests and not disabling checkPhase will result in an error.

jonringer commented on pkgs/development/python-modules/nsenter/default.nix:
```diff
@@ -0,0 +1,25 @@
+
+buildPythonPackage rec {
+  pname = "nsenter";
+  version = "0.2";
```
this hasn't seen an update in 4 years, it probably shouldn't go in nixpkgs

petabyteboy replied:
There have been new changes to the repository as recently as 16 days ago and there is no hint that it is unmaintained. If you want I can also package the latest master revision, but there haven't been significant changes, mostly because it still works fine as it is.

jonringer commented on pkgs/development/python-modules/nsenter/default.nix:
```diff
@@ -0,0 +1,25 @@
+  pname = "nsenter";
+  version = "0.2";
+
+  propagatedBuildInputs = [
+    contextlib2
+    argparse
+    flake8
+  ] ++ lib.optional (pythonOlder "3.4") [
+    pathlib
+  ];
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1jr5592pshzn0ck75xb4f8dmcya8vb7x8byp9j7991fy0g5ihsl7";
+  };
```
generally src is listed before inputs
```suggestion

  src = fetchPypi {
    inherit pname version;
    sha256 = "1jr5592pshzn0ck75xb4f8dmcya8vb7x8byp9j7991fy0g5ihsl7";
  };
  
  propagatedBuildInputs = [
    contextlib2
    argparse
    flake8
  ] ++ lib.optional (pythonOlder "3.4") [
    pathlib
  ];

```

petabyteboy replied:
Thanks, implemented for all new packages.

Commit:
python3Packages.cherrypy: 18.3.0 -> 18.5.0

The test `cherrypy/test/test_tools.py::ToolTests::testCombinedTools` is
failing since the python was updated to 3.7.7 and is now deselected.

Commit:
pythonPackages.nsenter: init at 0.2.0

Commit:
python3Packages.pylxd: init at 2.2.10

Commit:
python3Packages.pyshark: init at 0.4.2.9

petabyteboy commented:
I copy-pasted this from a private repo where I had packaged it a long time ago. I should have been more careful while doing that.

oxzi reviewed (commented):
I have tested the `pyshark` package successfully. The other packages I have not tried, but they look okay at first sight.

pstch commented:
I have tested the `pylxd` package successfully (actually wrote the exact same derivation before realising this PR already existed...). You can add me as maintainer for it if you want.
